owasp-top10-2017-apps/a2/saidajaula-monster/app/app.py

```diff
@@ -103,10 +100,7 @@ def login():
             return "Login failed! \n"
 
         cookie_dic = {"permissao": result[1], "username": form_username}
-        cookie = json.dumps(cookie_dic)
-        hash_cookie = hashlib.sha256(cookie.encode('utf-8')).hexdigest()
-        cookie_done = '.'.join([cookie,hash_cookie])
-        cookie_done = base64.b64encode(str(cookie_done).encode("utf-8"))
+        cookie_done = jwt.encode(cookie_dic, SECRET_KEY, algorithm='HS256')
```

Comment:
It would be nice to have an expiration time set here

Reply:
Hey, @gustavocovas. I created a function create_tokens() to create the access and refresh tokens, and I created a new path /refresh-token where the front-end application will call to generate a new access and refresh tokens.

The function refresh_token() redirect the user to the URL in referer header, unfortunately, how the application is running locally the browser doesn't send the referer. Therefore, you always will be sent to the /user.

```
referer = request.referrer
if referer != None:
    resp = redirect(referer)
else:
    resp = redirect("user")
```